File: McbEdu.Mentorias.ShopDemo.Infrascructure.Data/Repositories/ProductRepository.cs
using McbEdu.Mentorias.ShopDemo.Domain.Contexts.ProductContext.DTO;
using McbEdu.Mentorias.ShopDemo.Infrascructure.Data.Repositories.Interfaces;

namespace McbEdu.Mentorias.ShopDemo.Infrascructure.Data.Repositories;

public class ProductRepository : IExtendsProductRepository
{
    private readonly DataContext _dataContext;

    public ProductRepository(DataContext dataContext)
    {
        _dataContext = dataContext;
    }

    public Task AddAsync(Product entity)
    {
        return Task.FromResult(_dataContext.Products.Add(entity));   
    }

    public Task AddRangeAsync(Product entity)
    {
        throw new NotImplementedException();
    }

    public Task CommitChanges()
    {
        return Task.FromResult(_dataContext.SaveChanges());
    }

    public void Delete(Product entity)
    {
        throw new NotImplementedException();
    }

    public List<Product> GetAll()
    {
        throw new NotImplementedException();
    }

    public Task<Product?> GetByCode(string code)
    {
        return Task.FromResult(_dataContext.Products.Where(p => p.Code == code).FirstOrDefault());
    }

    public Product? GetByIdentifier(Guid identifier)
    {
        throw new NotImplementedException();
    }

    public Task<List<Product>> GetProductByPaginationFilteredByCode(int index, int offset)
    {
        throw new NotImplementedException();
    }

    public Task<List<Product>> GetProductByPaginationFilteredByDescription(int index, int offset)
    {
        throw new NotImplementedException();
    }

    public Task<List<Product>> GetProductByPaginationOrderringByDescription(int index, int offset)
    {
        throw new NotImplementedException();
    }

    public void Update(Product entity)
    {
        throw new NotImplementedException();
    }

    public void UpdateRange(List<Product> entities)
    {
        throw new NotImplementedException();
    }

    public Task<bool> VerifyEntityExistsAsync(string code)
    {
        return Task.FromResult(_dataContext.Products.Where(p => p.Code == code).Any());
    }

    public Task<bool> VerifyEntityExistsLocalAsync(string code)
    {
        return Task.FromResult(_dataContext.Set<Product>().Local.Where(p => p.Code == code).Any());
    }
}

File: src/main/java/com/mcbmentorias/shopdemo/infra/data/ProductRepository.java
package com.mcbmentorias.shopdemo.infra.data;

import com.mcbmentorias.shopdemo.domain.entities.product.Product;
import com.mcbmentorias.shopdemo.domain.repositories.IProductRepository;
import com.mcbmentorias.shopdemo.infra.data.adapters.ProductRepositoryAdapter;
import org.springframework.stereotype.Repository;

@Repository
public class ProductRepository implements IProductRepository {

    private final ProductRepositoryAdapter adapter;

    public ProductRepository(final ProductRepositoryAdapter adapter) {
        this.adapter = adapter;
    }

    @Override
    public Product save(final Product product) {
        return this.adapter.save(product);
    }

    @Override
    public Boolean checkIfProductExists(final String productCode) {
        return this.adapter.checkIfProductExists(productCode);

    }

    @Override
    public Product getByCode(final String code) {
        return this.adapter.getByCode(code);
    }
}
